### PR TITLE
Fix test for nightly Python (3.9)

### PR DESCRIPTION
**Pull request by PetterS**
Trying to get the nightly test in #530 to pass.

### CODE REVIEW

**Comment by PetterS** [comment]
OK, test passes now but feels way too brittle.

Should be enough to assert that the error starts with `sourcePath + ":8:"`

File: pyflakes/test/test_api.py [hunk 1]
```diff
@@ -447,19 +447,26 @@ def evaluate(source):
...
+            column = 11
+            whitespace_before_caret = 10
             if PYPY:
                 message = 'end of file (EOF) while scanning triple-quoted string literal'
-            else:
-                message = 'invalid syntax'
+            elif sys.version_info >= (3, 9):
+                message = 'invalid string prefix'
+                column = 52
+                whitespace_before_caret = 14
```
Comment (asottile) [inline pyflakes/test/test_api.py:458]: these offsets and error message seem like a bug, could you report this to upstream?  there is no line 14 and column 52 for starters and it's unclear why it's talking about a string prefix when the error is an unterminated string literal

Comment (PetterS) [reply pyflakes/test/test_api.py:458]: The reason it is talking about a string prefix is that that's the first error in encounters. The invalid string prefix is "quux".

Comment (PetterS) [reply pyflakes/test/test_api.py:458]: I don't know what to report to upstream. When I run the code, Python 3.9.0a5 gives me
```
  File "/mnt/c/Users/Admin/Desktop/syntaxerror.py", line 6
    '''quux'''
          ^
SyntaxError: invalid string prefix
```
which is correct.

And if I catch as print the exc_info I get
```
(<class 'SyntaxError'>, SyntaxError('invalid string prefix', ('<string>', 7, 50, "    '''\ndef bar():\n    pass\ndef baz():\n    '''quux'''\n")), <traceback object at 0x7fb1c824b340>)
```
This is identical to the system Python 3.6

I have no idea what the tests in this file are for, so can't have an opinion about what is reasonable here.

Comment (asottile) [reply pyflakes/test/test_api.py:458]: that's fine, I've gone ahead and made the report myself :) https://bugs.python.org/issue40335

Comment (PetterS) [reply pyflakes/test/test_api.py:458]: Aha! That's why I got the same with Python 3.6. Should have tried 3.8 as well

**Comment by asottile** [comment]
could you rebase this? it should be fixed upstream now

[here's a simpler patch you can use instead](https://github.com/python/cpython/pull/19619#issuecomment-616659821)

**Comment by PetterS** [comment]
Done!

But the Travis check does not seem to have run for some reason.

**Comment by asottile** [comment]
looks like the new PEG parser has broken these even further :(

**Comment by PetterS** [comment]
Drop these tests for nightly? Seems a little too fragile?

**Comment by asottile** [comment]
the tests are valuable, and have pointed out a bug in cpython

I think we'll probably `allow_failures: ...` for now though to allow patches to flow while we wait for cpython to fix things

**Comment by PetterS** [comment]
OK, where is that setting?

**Comment by asottile** [comment]
https://docs.travis-ci.com/user/customizing-the-build/#jobs-that-are-allowed-to-fail

**Pull request merged**